tests/torch/pruning/helpers.py

```diff
@@ -597,13 +597,91 @@ def forward(self, x):
         return self.fc(x)
 
 
+class SplitModel(nn.Module):
+    #         (input)
+    #            |
+    #         (conv1)
+    #            |
+    #         (chunk)
+    #        /      \
+    #    (conv2)  (conv3)
+    def __init__(self):
+        super().__init__()
+        self.conv1 = create_conv(1, 4, 1, 1)
+        self.conv2 = create_conv(4, 8, 1, 1)
+        self.conv3 = create_conv(4, 8, 1, 1)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        y1, y2 = torch.chunk(x, chunks=2, dim=-1)
+
+        y1 = self.conv2(y1)
+        y2 = self.conv3(y2)
+        return y1, y2
+
+
+class SplitMaskPropFailModel(nn.Module):
+    #         (input)
+    #            |
+    #         (conv1)
+    #            |
+    #         (chunk)
+    #        /      \
+    #    (conv2)  (conv3)
+    """
+    Weights have shape [N, C, C, W] and split dimension is not 1, but 2.
+    Mask propagation should fail because of inconsistency of number of channels (C)
+    and length of the resulting mask (C/2).
+    """
+    def __init__(self):
+        super().__init__()
+        self.conv1 = create_conv(1, 4, 4, 1)
+        self.conv2 = create_conv(4, 8, 1, 1)
+        self.conv3 = create_conv(4, 8, 1, 1)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        y1, y2 = torch.chunk(x, chunks=2, dim=2)
+
+        y1 = self.conv2(y1)
+        y2 = self.conv3(y2)
+        return y1, y2
```

Comment:
Shape of the x after `conv1` is `(1, 4, 5, 5)`, so  `generate_output_masks` from mask with shape `4` generates two masks with dims `[2, 3]` (because 5 is splitted on two.
1) In such case split should not propagate any masks through. You need to add check inside `generate_output_masks`
2) This case isn't a case with `[N, C, C, W]` tensor, please correct the model

Reply:
I understood it. I will change it like `self.conv1 = create_conv(1, 6, 3, 1)`. In this case, the split axis is `dim=2` and it seems like it could not proceed mask propagation more. Then in my thought, it seems necessary to make sure that the split axis fits with channel axis. What do you think?

Reply:
I agree with you. In ` generate_output_masks` should be a check like

```
input_mask.shape[0] == sum(output_edges_shapes)
```

Reply:
Oh, actually, case illustrated bellow will make pruning invalid because reshape mask propagation will just pass through existing masks, so pruning will be thinking that everything is ok 
![image](https://user-images.githubusercontent.com/74656388/186376648-f7b05f49-ee32-4c33-bc70-7d1269b6fc05.png)

Reply:
I added this case in the `test_model_pruning_analysis` with `SplitReshapeModel` and cover it in `_is_not_mixing_dim` in common/pruning/operations.py.